File: installer/src/main/java/com/teamscale/profiler/installer/HttpUrlTypeConverter.java
package com.teamscale.profiler.installer;

import okhttp3.HttpUrl;
import picocli.CommandLine;

public class HttpUrlTypeConverter implements CommandLine.ITypeConverter<HttpUrl> {
	@Override
	public HttpUrl convert(String value) {
		HttpUrl url = HttpUrl.parse(value);
		if (url == null) {
			throw new CommandLine.TypeConversionException("This is not a valid URL: " + value);
		}
		return url;
	}
}

File: installer/src/main/kotlin/com/teamscale/profiler/installer/HttpUrlTypeConverter.kt
package com.teamscale.profiler.installer

import okhttp3.HttpUrl
import okhttp3.HttpUrl.Companion.toHttpUrl
import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
import picocli.CommandLine

/** Helps Picocli to convert from String to HttpUrl.  */
class HttpUrlTypeConverter : CommandLine.ITypeConverter<HttpUrl> {
	override fun convert(value: String) =
		value.toHttpUrlOrNull() ?: throw CommandLine.TypeConversionException("This is not a valid URL: $value")
}
